### PR TITLE
FRONT/AR-21->dev

### PR DESCRIPTION
**Pull request by Beafowl-Pull**
* **Checklist before requesting a review**
- [x] The commits messages follows our guidelines
- [x] I have performed a self-review of my code
- [ ] I have documented my code
- [ ] Do we need to implement unit tests?

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of changes does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

## Summary by CodeRabbit

- **New Features**
  - Added a search functionality to the app.
  - Introduced a new user registration form with email, username, and password fields.

- **Documentation**
  - Updated `.gitignore` to exclude environment configuration files.

- **Style**
  - Enhanced the login and registration modules' visual design with CSS adjustments.
  - Expanded the color palette and utility classes in `tailwind.config.ts`.

- **Chores**
  - Modified development environment configuration to point to the new API endpoint.

### CODE REVIEW

**Comment by badidu09** [comment]
for the moment i'm okey but i found some issue like type of register button

**Review by badidu09 — APPROVED** [review]

**Review by badidu09 — APPROVED** [review]

**Review by romainpanno — DISMISSED** [review]

File: front/app/register/login.module.css [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  position: absolute;
+  width: 7%;
+  height: 15%;
+  left: 25%;
+  top: 10%;
+  background: #65517E;
+  opacity: 0.2;
+  transform: rotate(-45deg);
+}
+
```
Comment (Vivelis) [inline front/app/register/login.module.css:35]: Should be only one back to line at end of file.
```suggestion
}
```

**Review by Vivelis — CHANGES_REQUESTED** [review]

**Review by KitetsuK — DISMISSED** [review]

File: front/app/register/page.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        });
+
+        if (response.ok) {
+            // Redirect to home page
+            setLoginSuccess(true);
+        } else {
+            // Handle errors here
+            console.error('Error:', response);
+            setError('An error occurred, please retry');
+        }
```
[the remark this answers is not in the document]
Comment (romainpanno) [reply front/app/register/page.tsx:57]: ceate a methode for that

**Review by romainpanno — CHANGES_REQUESTED** [review]
Create a method for the login (in order to manipulqte the api)

**Review by romainpanno — APPROVED** [review]

**Review by KitetsuK — APPROVED** [review]

**Pull request merged**